### PR TITLE
Remove database subnet and add architecture diagram

### LINKED ISSUE #42: Add diagram to show AWS infrastructure layout
Use https://dev.to/rolfstreefkerk/how-to-setup-a-basic-vpc-with-ec2-and-rds-using-terraform-3jij as an example

### LINKED ISSUE #62: Remove database subnet
I think we don't need both the private and database subnets - the database subnet [can be used to expose the database to the public](https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws/5.0.0#public-access-to-rds-instances:~:text=%23%20...%20omitted%0A%7D-,Public%20access%20to%20RDS%20instances,-Sometimes%20it%20is), but is not recommended.

We could [create our own `aws_db_subnet_group`](https://developer.hashicorp.com/terraform/tutorials/aws/aws-rds#subnet-group) (using the private subnets), and use this when creating the RDS instance

### PR DESCRIPTION
**Pull request by p-j-smith**
Fixes #42 
Fixes #62 

- remove the database subnet and use only the private one. Create our own `aws_db_subnet_group` rather than using the `setup_vpc` module do it - this way we can put the private subnet ids in the group
- explicitly set the availability to put the RDS instance in (otherwise it will randomly choose)
- add a diagram of the infrastructure we're creating
- add some notes on the infrastructure / resources we're creating

### CODE REVIEW

File: assets/xnat-aws-architecture.png
Comment (milanmlft) [inline assets/xnat-aws-architecture.png:1]: 😍

File: provision/main.tf [hunk 1]
```diff
@@ -25,10 +25,9 @@ module "setup_vpc" {
 
   # NOTE: database subnets only without private subnets doesn't work
   # cfr. https://github.com/terraform-aws-modules/terraform-aws-vpc/issues/944
```
Comment (milanmlft) [inline provision/main.tf:27]:
```suggestion

```

This comment is now obsolete.

**Review by milanmlft — APPROVED** [review]
Looking very good! Thanks for adding more details regarding the Terraform setup.
Just a small suggestion to remove the now obsolete comment regarding database subnets.

File: provision/main.tf (hunk 1 above)
Comment (p-j-smith) [reply provision/main.tf:27]: well spotted!

**Pull request merged**